### PR TITLE
nav2_lifecycle_manager documentation

### PR DESCRIPTION
**Pull request by yathartha3**
## Basic Info
Added nav2_lifecycle_manager documentation.
Addressed #1317

### CODE REVIEW

File: nav2_lifecycle_manager/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#### Background on lifecycle enabled nodes
+The lifecycle enabled nodes in the navigation stack (e.g. map_server, planner_server, controller_server) have lifecycle functions namely on_configure(), on_activate(), on_deactivate(), on_cleanup(), on_shutdown(), and on_error(). Managed lifecycle nodes allow roslaunch to ensure that all components have been instantiated correctly before it allows any component to begin executing its behaviour. It will also allow nodes to be restarted or replaced on-line. More details about managed nodes can be found on [ROS2 Design website](https://design.ros2.org/articles/node_lifecycle.html).
```
Comment (SteveMacenski) [inline nav2_lifecycle_manager/README.md]: - the methods could be code, ei `on_configure()` not on_configure() 
- ros2launch
- Careful when you say "components" lifecycle nodes and component nodes aren't related

File: nav2_lifecycle_manager/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+#### nav2_lifecycle_manager
+The lifecycle manager is used to change the states of it's managed nodes in order to achieve a controlled startup, shutdown, reset, pause or resume of the navigation stack. 
+
+It presents a _"lifecycle_manager/manage_nodes"_ service, from which one can request the _startup_, _shutdown_, _reset_, _pause_, or _resume_ functions to be called. Based on this service request, the lifecycle manager calls the necessary lifecycle services in the lifecycle managed nodes. Currently, the RVIZ panel uses the _"lifecycle_manager/manage_nodes"_ sevice when user presses the buttons on the RVIZ panel (e.g.,startup, reset, shutdown, etc.).
+
+
+The lifecycle manager has a default nodes list as shown below:
+ ```
+ {"map_server", "amcl", "planner_server", "controller_server", "recoveries_server", "bt_navigator"}
```
Comment (SteveMacenski) [inline nav2_lifecycle_manager/README.md]: I'd remove the defaults. We'll all forget to update this as things change. Even just pending PRs have updates to this

File: nav2_lifecycle_manager/README.md (hunk 1 above)
Comment (yathartha3) [reply nav2_lifecycle_manager/README.md]: Done. Renamed components to nodes.

File: nav2_lifecycle_manager/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+### Background on lifecycle enabled nodes
```
Comment (mjeronimo) [inline nav2_lifecycle_manager/README.md]: Some edits:

**Background on lifecycle enabled nodes**
Using ROS2’s managed/lifecycle nodes feature allows the system startup to ensure that all required nodes have been instantiated correctly before they begin their execution. Using lifecycle nodes also allows nodes to be restarted or replaced on-line. More details about managed nodes can be found on ROS2 Design website. Several nodes in the navigation2 stack, such as map_server, planner_server, and controller_server, are lifecycle enabled. These nodes provide the required overrides of the lifecycle functions, on_configure(), on_activate(), on_deactivate(), on_cleanup(), on_shutdown(), and on_error(). 

**nav2_lifecycle_manager**
Navigation2’s lifecycle manager is used to change the states of the lifecycle nodes in order to achieve a controlled startup, shutdown, reset, pause or resume of the navigation stack. The lifecycle manager presents a lifecycle_manager/manage_nodes service, from which clients can invoke the startup, shutdown, reset, pause, or resume functions. Based on this service request, the lifecycle manager calls the necessary lifecycle services in the lifecycle managed nodes. Currently, the RVIZ panel uses this lifecycle_manager/manage_nodes service when user presses the buttons on the RVIZ panel (e.g.,startup, reset, shutdown, etc.).

In order to start the navigation stack and be able to navigate, the necessary nodes must be configured and activated. Thus, for example when startup is requested from the lifecycle manager's manage_nodes service, the lifecycle managers calls configure() and activate() on the lifecycle enabled nodes in the node list.

The lifecycle manager has a default nodes list for all the nodes that it manages. This list can be changed using the lifecycle manager’s “node_names” parameter. 

The diagram below shows an example of a list of managed nodes, and how it interfaces with the lifecycle

Comment (yathartha3) [reply nav2_lifecycle_manager/README.md]: Thanks, it looks much better now.

**Pull request merged**